src/Compilers/Extension/CompilerPackage.cs

```diff
@@ -70,12 +70,11 @@ private void WriteMSBuildFiles(string packagePath, string hiveName)
                     $@"<?xml version=""1.0"" encoding=""utf-8""?>
 <Project xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
   <!-- If we're not using the compiler server, set ToolPath/Exe to direct to the exes in this package -->
-  <PropertyGroup Condition=""'$(RoslynHive)' == '{hiveName}' and '$(UseSharedCompilation)' != 'true'"">
+  <PropertyGroup Condition=""'$(RoslynHive)' == '{hiveName}' and '$(UseSharedCompilation)' == 'false'"">
```

Comment:
So setting CscToolPath/VbcToolPath disables shared compilation? Then what's setting the path to the compiler otherwise?

Reply:
The targets should import the task, then the task uses Assembly.GetExecutingLocation to find the location of the server next to it.

Reply:
(The code for UsingTask is a half-dozen lines above)